apps/api/src/domain/models/user-management/utilities/coscrad-date.entity.ts

```diff
@@ -0,0 +1,19 @@
+import { InternalError } from '../../../../lib/errors/InternalError';
+import { MultilingualText } from '../../../common/entities/multilingual-text';
+import { AggregateCompositeIdentifier } from '../../../types/AggregateCompositeIdentifier';
+import { Aggregate } from '../../aggregate.entity';
+
+export class CoscradDate extends Aggregate {
+    protected validateComplexInvariants(): InternalError[] {
```

Comment:
You need some data fields and a constructor.
```ts
// what we really want is @DayOfMonth(...) that requires an integer from 1-31- add this to the validation lib
    @PositiveInteger(...) // for now
    readonly day: number;

    @ExternalEnum(...)// we need a new Month enum
    readonly month: Month;

    @Year(...)
    readonly year: number;
```

You might want to make a story for adding a new COSCRAD data type, but go ahead and do the enum now.

Reply:
[Here](https://www.pivotaltracker.com/story/show/187270562) is the link.